core/templates/dev/head/pages/state_editor/StateResponsesDirective.js

```diff
@@ -323,6 +323,9 @@ oppia.directive('stateResponses', [
 
                     EditorFirstTimeEventsService.registerFirstSaveRuleEvent();
 
+                    if (!$scope.tmpOutcome.missingPrerequisiteSkillId) {
```

Comment:
What is its value before being null?

Ideally we would set this value to null at the very outset, rather than having a possible case where it's "kinda null but not really".

Reply:
This is null when initialised, but I think it becomes undefined when an invalid sequence is entered, after validation by ng-pattern, which was causing the error in the issue.
So, here I am just checking if it's undefined and changing value to null.

Reply:
Hm, but I think it should never become undefined in the first place. Can we find a way to make that not happen?

Reply:
According to [this](https://stackoverflow.com/questions/23262953/scope-var-giving-undefined-in-angularjs) SO thread, it seems ng-pattern does that and to get the value stored in the variable is to use $viewValue. But here, we don't need the value, just that the varable shouldn't be undefined, which is not possible with ng-pattern.
Should I remove ng-pattern and use some validation in controller itself?

Reply:
Done.